packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -495,15 +495,11 @@ class GoTrueClient {
 
     final authResponse = AuthResponse.fromJson(response);
 
-    if (authResponse.session == null) {
-      throw AuthException(
-        'An error occurred on token verification.',
-      );
```

Comment:
This piece of code is mirroring [this](https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts#L673) js SDK code, but we actually need to allow empty session.

We shouldn't need the `!data` check equivalent here, because `_fetch.request` will throw if it did not have any response data.